ndarray/ndarray-tfjs/src/jsMain/kotlin/io.kinference.ndarray/extensions/MemoryExtension.kt

```diff
@@ -16,10 +27,22 @@ fun <T : NDArrayTFJS> tidyNDArrays(fn: () -> Array<T>): Array<T> {
 }
 
 @Suppress("UNCHECKED_CAST")
-fun <T : NDArrayTFJS> tidyNDArray(fn: () -> T): T {
+suspend fun <T : NDArrayTFJS> tidyNDArray(fn: suspend () -> T): T {
     val rawOutput = tidy {
         val output = fn()
         return@tidy arrayOf(output.tfjsArray)
     }.first()
     return rawOutput.toNDArray() as T
 }
+
+suspend fun scopedRun(start: () -> Unit, end: () -> Unit, fn: suspend () -> Array<ArrayTFJS>): Array<ArrayTFJS> {
+    start()
+    try {
+        val res = fn()
+        end();
+        return res;
+    } catch (e: Exception) {
+        end()
```

Comment:
finally?

Reply:
That's function was fully copied from TFJS sources. I think `end()` must be called before `return res`